src/Messaging/MessageBuffer.php

```diff
@@ -60,6 +72,31 @@ function __construct(
         $this->onControl = $onControl ?: function() {};
 
         $this->leftovers = '';
+
+        $memory_limit = \trim(\ini_get('memory_limit'));
+        $memory_limit_bytes = 0;
+        if ($memory_limit !== '') {
+            $shifty = ['k' => 0, 'm' => 10, 'g' => 20];
+            $multiplier = strlen($memory_limit) > 1 ? substr(strtolower($memory_limit), -1) : '';
+            $memory_limit = (int)$memory_limit;
+            $memory_limit_bytes = in_array($multiplier, array_keys($shifty), true) ? $memory_limit * 1024 << $shifty[$multiplier] : $memory_limit;
+        }
+        if ($maxMessagePayloadSize === null) {
+            $maxMessagePayloadSize = $memory_limit_bytes / 4;
+        }
+        if ($maxFramePayloadSize === null) {
+            $maxFramePayloadSize = $memory_limit_bytes / 4;
+        }
```

Comment:
I'm thinking we should also have a floor in case the memory configuration is tiny. What ever the largest message sent in the AB tests are. Off the top of my head I think it's 4MB.

Reply:
I picked 1/4 because, reasonably (I haven't tested it), that is the largest the library could handle. It has to have enough memory to copy the payload out on the `getPayload` call - which will require more than 2 times the size of the message to do.

Reply:
I think we can ignore my first comment...It looks like the largest AB test is 16MB. My concern was that is the PHP memory limit was set too low the AB tests couldn't run. The default PHP memory limit is 128MB which would constitute 1/4 being 32MB, so I think we're ok.